doc/dev/adr/0023-large-message-chunking.md

```diff
@@ -0,0 +1,120 @@
+﻿# ADR 23: Large Message Chunking in MQTT Protocol
+
+## Status
+
+Proposed
+
+## Context
+
+The MQTT protocol has inherent message size limitations imposed by brokers and network constraints. Azure IoT Operations scenarios often require transmitting payloads that exceed these limits (e.g., firmware updates, large telemetry batches, complex configurations). Without a standardized chunking mechanism, applications must implement their own fragmentation strategies, leading to inconsistent implementations and interoperability issues.
+
+## Decision
+
+We will implement sdk-level message chunking as part of the Protocol layer to transparently handle messages exceeding the MQTT broker's maximum packet size.
+
+**The chunking mechanism will**:
+
+- Be enabled/disabled by a configuration setting.
```

Comment:
What is the default if not set?  I think it should be disabled for compat concerns esp. if a non-SDK is receiving the PUBLISH messages.